src/content/blog/astro-240.mdx

```diff
@@ -0,0 +1,122 @@
+---
+title: "Astro 2.4"
+description: ""
+publishDate: "May 4, 2023"
+authors:
+  - matthew
+coverImage: "/src/content/blog/_images/astro-220/blog-hero-2_2-release.jpg"
+socialImage: "/src/content/blog/_images/astro-220/og-image-2_2-release.jpg"
+lang: "en"
+---
+
+Astro 2.4 just dropped, and features:
+
+- **[Stronger scoping for CSS](#scopedstylestrategy)**: A new opt-in flag `scopedStyleStrategy` allows you to configure your Astro components to use a high specificity for its styles.
+- **[Improved &lt;Code /&gt; component](#code--component-improvements)**: Upgraded Shiki and support for inlining.
+- **[SSR support in sitemaps](#ssr-in-sitemaps)**: Known routes are now built into a sitemap during the build.
+- **[Middleware (experimental)](#middleware-experimental)**: Define code that runs before page components and endpoints, to intercept requests and modify responses.
+- **[CSS inlining (experimental)]()**: Allows you to configure when styles should be inlined via a `<style>` tag.
+
+## scopedStyleStrategy
+
+Start in 1.0 Astro uses the `:where` pseudo-selector for scoped Astro component styles. This approach means your styles have the same specificity as if they were written in plain CSS. The downside of this approach is that you need to be conscience of the selectors used in components.
+
+As users _usually_ expect their component styles to override global styles we've added the `scopedStyleStrategy` option which can be used to switch on a high-specificity strategy:
+
+```js
+import { defineConfig } from "astro/config"
+
+export default defineConfig({
+	scopedStyleStrategy: "class",
+})
+```
+
+This strategy will likely be used as the default in 3.0.
+
+## &lt;Code /&gt; component improvements
+
+The &lt;Code /&gt; has a few new improvements:
+
+- Upgrade to **Shiki 0.14** - Astro was previously on Shiki 0.11; this upgrade comes with numerous improvements, most notably new themes and language support.
+- new `inline` prop prevents a wrapping `<pre>` from being added. This is useful if you want syntax highlighting inline in a paragraph.
+
+## SSR in sitemaps
+
+The `@astrojs/sitemaps` package has been updated to include support for server-side rendering. Previously sitemaps were only supported when using `output: 'static'` because all dynamic routes are not known at build time.
+
+The new version will now output a sitemap for _known_ routes. Dynamic routes are excluded. No changes are needed to your configuration in order to enable this feature.
```

Comment:
```suggestion
## SSR in sitemaps

The `@astrojs/sitemaps` package has been updated to include support for server-side rendering. Previously, sitemaps were only supported when using `output: 'static'` because dynamic routes are unknown at build time.

The new version can now also be used with `output: 'ssr'` and will output a sitemap for _known_ (i.e. static) routes only. Dynamic routes are excluded. No changes are needed to your configuration in order to enable this feature.
```